generator/src/pkg/generate/generate.go

```diff
@@ -123,68 +141,81 @@ func Unique(strSlice []string) int {
 			list = append(list, entry)
 		}
 	}
-	return len(list)
+	return list
 }
 
-// Parse microservice chain and cluster file, and return a mapped placement struct
-func Parse(chainFilename, clusterFilename string) (m map[string]map[string][]string, p Placement) {
-
-	// filePath := fmt.Sprintf("chains/%s", chainFilename)
-
-	chainFile, err := os.Open(chainFilename)
+// Parse microservice config file, and return a config struct
+func Parse(configFilename string) (Config, []string) {
+	configFile, err := os.Open(configFilename)
+	configFileByteValue, _ := ioutil.ReadAll(configFile)
 
 	if err != nil {
 		fmt.Println(err)
 	}
 
-	chainFileByteValue, _ := ioutil.ReadAll(chainFile)
-
-	// clusterFilePath := fmt.Sprintf("clusters/%s", clusterFilename)
-
-	clusterFile, e := os.Open(clusterFilename)
+	var loaded_config Config
+	json.Unmarshal(configFileByteValue, &loaded_config)
+	for i := 0; i < len(loaded_config.Services); i++ {
+		services = append(services, loaded_config.Services[i].Name)
+		for j := 0; j < len(loaded_config.Services[i].Clusters); j++ {
+			clusters = append(clusters, loaded_config.Services[i].Clusters[j].Cluster)
+		}
+		for k := 0; k < len(loaded_config.Services[i].Endpoints); k++ {
+			endpoints = append(endpoints, loaded_config.Services[i].Endpoints[k].Name)
+		}
 
-	if e != nil {
-		fmt.Println(e)
 	}
 
-	clusterFileByteValue, _ := ioutil.ReadAll(clusterFile)
+	fmt.Println("All clusters: ", Unique(clusters))
+	fmt.Println("Number of clusters: ", len(Unique(clusters)))
+	fmt.Println("---------------")
+	fmt.Println("All Services: ", Unique(services))
+	fmt.Println("Number of services (unique): ", len(Unique(services)))
+	fmt.Println("---------------")
+	fmt.Println("All endpoints: ", Unique(endpoints))
+	fmt.Println("Number of endpoints: ", len(Unique(endpoints)))
+	return loaded_config, clusters
+}
```

Comment:
Could you add a printscreen of what this returns to show the parser is actually performing well?

Reply:
Don't the current printings show that it is performing well? because the printed slices are filled with the data from the loaded config!

Reply:
At first, I misunderstood, but after the discussion here is the output of sample json that we have:
```
./generator.sh input/new_description.json 10
deleting previous manifest files
generating kubernetes manifest files
All clusters:  [cluster_1]
Number of clusters:  1
---------------
All Services:  [service_1]
Number of services (unique):  1
---------------
All endpoints:  [/end_1]
Number of endpoints:  1

```